### DISCUSSION TITLE
read CSV file, assuming same format as OpenGammaDetector serial out?

### DISCUSSION THREAD

**Discussion by jbeale1** [Open-Gamma-Project/Gamma-MCA#225] [Ideas]
With the addition of one opamp to invert the negative-going pulse from a PMT device, I was able to use the OpenGamma board to record its output. (This is just a plastic scintillator, not a real crystal so the spectrum isn't much, but it does work at least.) Because I don't have my computer at my workbench, I recorded the output from the OpenGamma board with a USB-Serial Terminal app on my phone, then transferred that file to my computer, and replaced semicolons with commas so I could read in the CSV file into the web app. This works and it does show the total count, but I wonder if it would be generally useful to have a feature that assumes the CSV file records one line per second, so when reading such a CSV file it could also show the elapsed time, based on the number of lines in the file? If so it could also show the "time evolution" of the signal, in an analogous way. Or maybe this feature is already present and I just wasn't aware of it?
https://photos.app.goo.gl/CZrsSpEFRuY8oLdk7

**Comment by NuclearPhoenixx** [comment]
There is currently no such feature. Sounds like an interesting idea, however, I'm not too sure on how to handle the time resolution in that case. Some people might have built their own Arduino spectrometer and output at different rates, so assuming just one line per second doesn't sound like a safe way to do it.

Anyways, there is a feature built into the Open Gamma Detector that's exactly what you need: [Standalone Data Recording](https://github.com/OpenGammaProject/Open-Gamma-Detector/tree/main/software#standalone-data-recording) (see also the [Hackaday post](https://hackaday.io/project/185211-all-in-one-gamma-ray-spectrometer/log/227031-fw-420-recording-to-flash)). You set a time to record, let the device handle the measurement and then just print the final output file once it's done. Just copy and paste the serial output into a file on your PC or smartphone and then the file is already formatted in a way so that you can directly import it into Gamma MCA. It doesn't allow for the time evolution chart, but it saves the recording time and some other meta data. This is just the standard JSON file format that Gamma MCA natively uses. Let me know if you have any questions about it!

On that occation, may I know how long your PMT pulses are in the time domain? It's really cool that you got it working, just curious since they tend to be a lot longer than the SiPM pulses.

**Reply by jbeale1** [reply]
Thanks for the note about standalone usage, I did not realize that was an option, I will try it. My scope says the PMT pulse after going through my inverting opamp, right as it goes into the OpenGammaDetector board is typically just under 1 microsecond in width, although the largest pulses are longer.  I am using a cheap, slow 5 MHz BW opamp and I could make it faster, but I thought it was good enough to start with.  Scope photo: https://photos.app.goo.gl/Hc5nbjNowQ6t4kceA

**Reply by NuclearPhoenixx** [reply]
Great, thanks. That should work just fine with the board 👍🏻
